From bd15f47d883f34d9a540ee55c79d9df3abcc6648 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Tue, 12 Jun 2018 14:25:48 +0300
Subject: [PATCH 3976/5242] MLK-18543-1: drm/bridge: nwl-dsi: Add pm_runtime
 support

commit  4128b6960677d700827f32a423726d1237b1e65d from
https://source.codeaurora.org/external/imx/linux-imx.git

Add support for pm_runtime, so that the assigned power domain state will be
changed accordingly.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/nwl-dsi.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/bridge/nwl-dsi.c b/drivers/gpu/drm/bridge/nwl-dsi.c
index 494379d..c6a8caa 100644
--- a/drivers/gpu/drm/bridge/nwl-dsi.c
+++ b/drivers/gpu/drm/bridge/nwl-dsi.c
@@ -1009,6 +1009,8 @@ static void nwl_dsi_bridge_enable(struct drm_bridge *bridge)
 		return;
 	}
 
+	pm_runtime_get_sync(dev);
+
 	ret = devm_request_irq(dev, dsi->irq,
 			       nwl_dsi_irq_handler, 0, IRQ_NAME, dsi);
 	if (ret < 0) {
@@ -1086,6 +1088,8 @@ static void nwl_dsi_bridge_disable(struct drm_bridge *bridge)
 
 	devm_free_irq(dev, dsi->irq, dsi);
 
+	pm_runtime_put_sync(dev);
+
 	dsi->enabled = false;
 }
 
@@ -1166,6 +1170,8 @@ static int nwl_dsi_probe(struct platform_device *pdev)
 		return -EINVAL;
 	}
 
+	pm_runtime_enable(dev);
+
 	dsi->no_clk_reset = of_property_read_bool(dev->of_node, "no_clk_reset");
 
 	dsi->dev = dev;
@@ -1188,6 +1194,8 @@ static int nwl_dsi_remove(struct platform_device *pdev)
 
 	drm_bridge_remove(&dsi->bridge);
 
+	pm_runtime_disable(&pdev->dev);
+
 	return 0;
 }
 
-- 
1.7.9.5

